engine/xml-config/src/main/java/org/datacleaner/job/JaxbJobReader.java

```diff
@@ -533,13 +534,22 @@ private void wireOutputDataStreams(final Map<ComponentType, ComponentBuilder> co
 
                 for (int i = 0; i < sourceColumnTypes.size(); i++) {
                     final ColumnType sourceColumnPath = sourceColumnTypes.get(i);
-                    final String outputStreamColumnPathName =
-                            getOutputStreamColumnPath(sourceColumnTypes.get(i).getPath(), componentType,
-                                    componentBuilder, i);
+                    final List<MetaModelInputColumn> findSourceColumn = sourceColumns.stream().filter(
+                            sourceCol -> sourceCol.getName().equals(sourceColumnPath.getPath())).collect(Collectors
+                                    .toList());
+                    final String outputStreamColumnPathName;
+                    if (findSourceColumn.size() > 0) {
+                        outputStreamColumnPathName = getOutputStreamColumnPath(findSourceColumn.get(0).getName(),
+                                componentType, componentBuilder, i);
+                    } else {
+                        outputStreamColumnPathName = getOutputStreamColumnPath(sourceColumns.get(i).getName(),
+                                componentType, componentBuilder, i);
```

Comment:
This seems to work in testing, but it's unclear to me... If you find the column from `sourceColumns` by name, you take that one, otherwise you take the same from sourceColumns, but by index instead. But if the index always matches, why search for the name?